### ISSUE TITLE
Property names from extended generic interfaces are not checked for quotes

### ISSUE THREAD

**Issue by mjarr** [rhys-vdw/ts-auto-guard#167]
## Source
```ts
export interface Generic<T = number> {
  "a(b)": T;
}

export interface NotGeneric {
  "a(b)": string;
}

/** @see {isExtendedGeneric} ts-auto-guard:type-guard */
export interface ExtendedGeneric extends Generic<string> {
  "c": string;
}

/** @see {isExtendedNotGeneric} ts-auto-guard:type-guard */
export interface ExtendedNotGeneric extends NotGeneric {
  "c": string;
}
```
## Result
```ts
import { ExtendedGeneric, ExtendedNotGeneric } from "./test";

export function isExtendedGeneric(obj: any, _argumentName?: string): obj is ExtendedGeneric {
    return (
        (obj !== null &&
            typeof obj === "object" ||
            typeof obj === "function") &&
        typeof obj.a(b) === "string" &&
        typeof obj["c"] === "string"
    )
}

export function isExtendedNotGeneric(obj: any, _argumentName?: string): obj is ExtendedNotGeneric {
    return (
        (obj !== null &&
            typeof obj === "object" ||
            typeof obj === "function") &&
        typeof obj["a(b)"] === "string" &&
        typeof obj["c"] === "string"
    )
}
```
## Expectation
`"a(b)"` should be accessed using index syntax in both cases. If a proper fix is difficult then how about introducing a new option forcing index syntax? The `command` part of the JSDoc tag looks like a good place for it as it would reduce the scope only to problematic guards.

**Comment by rhys-vdw** [comment]
Thanks for the report.

>  If a proper fix is difficult then how about introducing a new option forcing index syntax?

There is no reason to make the fix to this bug optional. It should work in all cases, ideally correcting the check for whether the string is a valid property name.

My feeling is that ts-morph should somehow expose functionality to correct differentiate the two cases, but when I last touched it wasn't obvious. Currently I believe we're relying on whether the property name from ts-morph has quotes within the string, which clearly is falling down here. I would start by creating a question issue on ts-morph to see if this functionality exists.

Given that this has been an ongoing problem (#123 #118), I think that always using an indexer check may be appropriate if no simple solution can be found.

Thought: It's also possible that generating an AST instead of raw strings might solve this elegantly, but that implies a rewrite of the whole program.

**Comment by rhys-vdw** [comment]
Closed by #198
